### PR TITLE
Implements navigator for bid flow.

### PR DESCRIPTION
**Pull request by ashfurrow**
This pull request is for https://artsyproduct.atlassian.net/browse/PURCHASE-41 . It replaces the "hello, world" component for the bid flow with a [navigator component](https://facebook.github.io/react-native/docs/navigatorios.html) that will give us the push/pop navigation stack that iOS developers are used to. The way it works is that each component rendered with it will get passed props to manipulate the navigation stack ([here is an example](https://github.com/artsy/emission/blob/master/src/lib/Components/Consignments/Screens/Welcome.tsx#L27-L30)). 

This PR also changes `BidFlow.tsx` to not use default exports, which we're trying to move away from.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Bidding/Screens/__tests__/BidFlow-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ashfurrow** [comment]
Oh hey whoops, now that that the component isn't hello world it should have tests.

File: src/lib/Components/Bidding/Screens/__tests__/BidFlow-tests.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from "react"
+import "react-native"
+import * as renderer from "react-test-renderer"
+
+import { BidFlow } from "../BidFlow"
+
+it("renders properly", () => {
+  const bg = renderer.create(<BidFlow saleArtworkID="5aada729139b216c0bf18103" />).toJSON()
+  expect(bg).toMatchSnapshot()
```
Comment (ashfurrow) [inline src/lib/Components/Bidding/Screens/__tests__/BidFlow-tests.tsx]: Okay! So now that the `BidFlow` component is going to _use_ the `SelectMaxBid` component, this poses a challenge with #1014. We'll need to modify the tests here to have an initial GraphQL state once that PR is merged. I'm happy to take care of that, or pair. 

/cc @sepans

**Comment by ashfurrow** [comment]
I've WIP'd this until #1014 gets merged, since it'll have implications on this PR (that I mention above).

File: src/lib/Components/Bidding/Screens/BidFlow.tsx [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
-export default class BidFlow extends React.Component<BidFlowProps> {
+export class BidFlow extends React.Component<BidFlowProps> {
   render() {
-    return <SerifText>This will be the home of the new bidding flow.</SerifText>
+    return (
+      <NavigatorIOS
+        navigationBarHidden={true}
+        initialRoute={{
+          component: SelectMaxBid,
+          title: "Your max bid", // title is required, though we won't likely use/display it.
```
Comment (yuki24) [inline src/lib/Components/Bidding/Screens/BidFlow.tsx]: I wonder if there's any way to pass this `title` in to each component so we don't have to hard-code them (though I think there's some trede-off here)

File: src/lib/Containers/BidFlow.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    )
+  }
+}
+
+export default createFragmentContainer(
+  BidFlow,
+  graphql`
+    fragment BidFlow_sale_artwork on SaleArtwork @relay(mask: false) {
+      bid_increments
+      ...SelectMaxBid_sale_artwork
```
Comment (ashfurrow) [inline src/lib/Containers/BidFlow.tsx]: Okay so the two lines above are a problem. `...SelectMaxBid_sale_artwork` should spread on that fragment, which includes `bid_increments`. But if we remove `bid_increments` from the line above, they aren't fetched. It is as though the `SelectMaxBid_sale_artwork` fragment isn't being spread correctly.

I'll keep looking into this, but I haven't found any promising leads. @sarahscott @orta any ideas here?

**Comment by ashfurrow** [comment]
Okay so after looking with @l2succes, our hypothesis is that we've hit an edge case between Relay and GraphQL for lists of non-object scalars. `bid_increments` is `GraphQLList(GraphQLFloat)`, which is odd, and we were planning to move away from that anyway, so I'm going to put this PR on hold while I make that change to Metaphysics.

**Comment by ashfurrow** [comment]
Okay, this is ready for review @yuki24. Unfortunately I couldn't get the Relay fragments to work as intended, but made a workaround suggested by @l2succes here: https://github.com/artsy/emission/pull/1015/commits/8cdc09b460c35d4419bec563c858b5345c432c5e Once this is merged, we can look at doing things the "right way", but this PR is blocking a lot of work, so I'd prefer to get it merged and improve it later. 

Here's an update screenshot of what's going on:

screen shot 2018-04-20 at 12 11 04 pm

**Comment by sweir27** [comment]
Coool! The workaround seems like a 👍plan to unblock other work, thanks!

**Comment by ashfurrow** [comment]
Okay, going to self-merge this.

**Pull request merged**